desktop/core/src/desktop/js/apps/storageBrowser/StorageBrowserPage/StorageBrowserTable/StorageBrowserPagination/StorageBrowserPagination.tsx

```diff
@@ -0,0 +1,125 @@
+// Licensed to Cloudera, Inc. under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  Cloudera, Inc. licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+import React from 'react';
+import { Button, Dropdown } from 'antd';
+import type { MenuProps } from 'antd';
+import { i18nReact } from '../../../../../utils/i18nReact';
+
+import PageFirstIcon from '@cloudera/cuix-core/icons/react/PageFirstIcon';
+import PagePreviousIcon from '@cloudera/cuix-core/icons/react/PagePreviousIcon';
+import PageNextIcon from '@cloudera/cuix-core/icons/react/PageNextIcon';
+import PageLastIcon from '@cloudera/cuix-core/icons/react/PageLastIcon';
+import DropdownIcon from '@cloudera/cuix-core/icons/react/DropdownIcon';
+
+import { PageStats } from '../../../../../reactComponents/FileChooser/types';
+import './StorageBrowserPagination.scss';
+
+interface StorageBrowserPaginationProps {
+  onPageNumberChange: (pageNumber: number) => void;
+  onPageSizeChange: (pageSize: number) => void;
+  pageSize: number;
+  pageSizeOptions?: number[];
+  pageStats: PageStats;
+}
+
+const defaultProps = {
+  currentPage: 1,
+  pageSizeOptions: [15, 30, 45, 60, 100, 200, 1000]
+};
+
+const StorageBrowserPagination: React.FC<StorageBrowserPaginationProps> = ({
```

Comment:
Currently I've made it a storage browser specific component because it gets all the data directly from the api. But yes it could be made into a separate component

Reply:
I think we should do that. Pagination is a common and reusable component which we will need in more places moving forward. We should try to make these kind of components API independent, and decouple the UI stuff from the backend stuff. I think states (like you have done) and data/api specific processing should be lifted up to the parent component.

Reply:
That said, we should also add unit tests :-)